addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -720,8 +720,9 @@ export const editorCommands = {
         }));
         referenceRow[beforeOrAfter](newRow);
         newRow.style.height = referenceRowHeight + 'px';
-        // Preserve the width of the columns (applied only on the first row).
-        if (getRowIndex(newRow) === 0) {
+        // Preserve the width of the columns (applied only on the first row and when any cell has fixed width).
+        const withFixedWidthCell = [...cells].map(cell => !cell.style.width).some(el => el === false);
+        if (getRowIndex(newRow) === 0 && withFixedWidthCell) {
```

Comment:
I would remove the use of `referenceRowWidths`.
That logic seems flawed.
If there was no width on the columns, it means that the user has never resized the width of a column, we should not retrieve the `clientWidth` and force a particular width.

Without the use of `referenceRowWidths`, the code could be rewritten like so:

```js
if (getRowIndex(newRow) === 0) {
            let columnIndex = 0;
            for (const newColumn of newRow.children) {
                newColumn.style.width = cells[columnIndex].style.width;
                cells[columnIndex].style.width = '';
                columnIndex++;
            }
        }
```

@Zynton are you ok with those changes or is there a hidden reason behind using `clientWidth`?

Reply:
It's not clear and I can't find a trace of a discussion about it. I do remember there being a reason though but I guess we can try removing it and test extensively to make sure nothing breaks. If nothing breaks we're good...

Reply:
Thanks guys! I will adapt the code, test it and put it back to TestIN so we make sure not to break anything :pray:

Reply:
@Goaman I found another related issue that, the table is shown improperly after we change the width of a column (spill out the form)
![2023-12-12_10-18](https://github.com/odoo/odoo/assets/23055668/c4edb6e9-816b-4786-8053-6bb886db08af)

I will try to find a solution for this and push it to this PR